### PR TITLE
Rename stores

### PR DESCRIPTION
**Pull request by justinmchase**
I'm intending to add a new abstraction for s3 file storage (#135). This is simply a straight up refactor to rename the stores based on their purpose rather than the names of the kinds of storage. In other words, instead of `fs` and `level` the stores are renamed to `block` and `data`.

The s3 storage type will be in another repo and will be very similar to the fs datastore but saving into s3.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -131,17 +122,21 @@ class IpfsRepo {
...
-          datastore: flatfs
-        }, {
-          prefix: new Key('/'),
-          datastore: new LevelStore(path.join(this.path, levelDirectory), {
-            db: this.options.level
-          })
-        }])
+        const DataStore = this.options.dataStore
+        const dataStore = new DataStore(path.join(this.path, dataStoreDirectory), this.options.dataStoreOptions)
+        log(dataStore)
```
Comment (dignifiedquire) [inline src/index.js]: this should be removed

File: test/node.js [hunk 2]
```diff
@@ -13,35 +13,28 @@ const expect = chai.expect
...
-    opts: undefined,
-    init: false
-  }, {
-    name: 'memory',
-    opts: {
-      fs: require('interface-datastore').MemoryDatastore,
-      level: require('memdown'),
-      lock: 'memory'
+  const repos = [
+    {
```
Comment (dignifiedquire) [inline test/node.js]: please don't change code style, this makes it hard to read the diff

File: test/node.js [hunk 3]
```diff
@@ -13,35 +13,28 @@ const expect = chai.expect
 const IPFSRepo = require('../src')
 
 describe('IPFS Repo Tests on on Node.js', () => {
-  const repos = [{
-    name: 'default',
-    opts: undefined,
-    init: false
```
Comment (dignifiedquire) [inline test/node.js]: why was this removed?

File: test/node.js [hunk 4]
```diff
@@ -13,35 +13,28 @@ const expect = chai.expect
...
-  }, {
-    name: 'memory',
-    opts: {
-      fs: require('interface-datastore').MemoryDatastore,
-      level: require('memdown'),
-      lock: 'memory'
```
Comment (dignifiedquire) [inline test/node.js]: why was this removed?

File: test/node.js [hunk 5]
```diff
@@ -13,35 +13,28 @@ const expect = chai.expect
...
     const testRepoPath = path.join(__dirname, 'test-repo')
     const date = Date.now().toString()
     const repoPath = testRepoPath + '-for-' + date
-
     const repo = new IPFSRepo(repoPath, r.opts)
-
     before((done) => {
       series([
-        (cb) => {
-          if (r.init) {
```
Comment (dignifiedquire) [inline test/node.js]: why are you removing this?

File: src/index.js [hunk 6]
```diff
@@ -34,33 +33,25 @@ class IpfsRepo {
...
     this.closed = true
     this.path = repoPath
-    this.options = Object.assign({
-      sharding: true,
-      lock: 'fs'
-    }, options)
-    this._fsOptions = Object.assign({}, options.fsOptions)
-    const FsStore = this.options.fs
-    this._fsStore = new FsStore(this.path, Object.assign({}, this._fsOptions, {
-      extension: ''
```
Comment (dignifiedquire) [inline src/index.js]: you can't just remove options that are important

File: src/index.js [hunk 7]
```diff
@@ -34,33 +33,25 @@ class IpfsRepo {
...
-    }))
+    this.options = Object.assign({ lock: 'memory', sharding: true }, options || defaultOptions)
```
Comment (dignifiedquire) [inline src/index.js]: `lock` still needs to be `fs` not `memory`

File: src/index.js (hunk 6 above)
Comment (diasdavid) [inline src/index.js]: I fixed this in #138

File: src/index.js (hunk 7 above)
Comment (justinmchase) [inline src/index.js]: Question about this. The `fs` lock appears to not use the datastore but to use fs api's directly. Supposing you were going to use a virtual file storage, and not the local file system, such as s3 would you still want a file system lock? It's not that the file system is completely off limits from writing to but the machine can get taken down and back up and the file system is ephemeral with the instance. Should the lock file also be stored on external storage or is it tied to the life cycle of the process?

Is its purpose to prevent other apps from writing the same files and if so shouldn't the lock actually in the one of the stores?

File: src/index.js (hunk 6 above)
Comment (justinmchase) [inline src/index.js]: Yup, that was missed when moving options into default options objects.

File: test/node.js (hunk 5 above)
Comment (justinmchase) [inline test/node.js]: For a bad reason. Basically I thought it was an inconsistency in the way the stores were getting created and a bug in this test but now I see that the tests were failing because I dropped the `extension: ''` from the store options which was making the fs based tests fail because it couldn't find the files copied in the before step. Adding the extension back and reverting basically this whole file resolves it.

File: src/index.js (hunk 1 above)
Comment (justinmchase) [inline src/index.js]: 👍

**Comment by justinmchase** [comment]
I updated my branch by adding the `extension: ''` option back onto the BlockStore in the constructor which also allowed me to revert the init changes to the tests.
